lib/rake/build.rake

```diff
@@ -7,9 +7,9 @@ namespace :build do
   desc 'Builds apps.'
   task :apps do
     Dir.chdir(apps_dir) do
-      # Only rebuild if apps contents have changed since last build.
+      # Only rebuild if apps contents or shared_constants have changed since last build.
       commit_hash = apps_dir('build/commit_hash')
-      if !RakeUtils.git_staged_changes?(apps_dir) &&
+      if !apps_changed? &&
         File.exist?(commit_hash) &&
         File.read(commit_hash) == RakeUtils.git_folder_hash(apps_dir)
```

Comment:
I'm not sure this is right - you've modified the check for _staged_ apps changes on the machine, but I think it's usually the `git_folder_hash` comparison that detects changes we commit in a PR and merge into the staging branch.

Reply:
Of course. Good catch. No wonder this was so easy to mimic on my local environment. I'll take another look at it later...

Reply:
:( This looks more complicated now since we have a single `commit_hash`, that is based on the `apps` directory. We'll need to either do some rearranging so that `apps` and `shared_constants` share a common base, or modify the commit hash to inspect multiple directories (I'm not sure we can), or use multiple commit hashes. 

This is a larger work item, so I'm going to abandon it for now and make a note in the tech debt backlog.